osu.Game/Beatmaps/Drawables/Cards/CollapsibleButtonContainer.cs

```diff
@@ -86,9 +80,6 @@ public CollapsibleButtonContainer(APIBeatmapSet beatmapSet)
                     RelativeSizeAxes = Axes.Both,
                     Anchor = Anchor.CentreRight,
                     Origin = Anchor.CentreRight,
-                    // workaround for masking artifacts at the top & bottom of card,
-                    // which become especially visible on downloaded beatmaps (when the icon area has a lime background).
-                    Padding = new MarginPadding { Vertical = 1 },
```

Comment:
I don't really want to defend this hack too much, but I don't feel that the screenshots in the OP do it justice as to how bad this can look. The worst of it is easily when the card is hovered:

![image](https://user-images.githubusercontent.com/20418176/195669967-d1271520-d5bc-4684-a6dc-a907b19be83d.png)

That said, you can do something like the following to attempt to mask it:

<details>
<summary>patch</summary>

```diff
diff --git a/osu.Game/Beatmaps/Drawables/Cards/CollapsibleButtonContainer.cs b/osu.Game/Beatmaps/Drawables/Cards/CollapsibleButtonContainer.cs
index f70694bdda..6e32454d99 100644
--- a/osu.Game/Beatmaps/Drawables/Cards/CollapsibleButtonContainer.cs
+++ b/osu.Game/Beatmaps/Drawables/Cards/CollapsibleButtonContainer.cs
@@ -77,7 +77,7 @@ public CollapsibleButtonContainer(APIBeatmapSet beatmapSet)
                 downloadTracker,
                 background = new Container
                 {
-                    RelativeSizeAxes = Axes.Both,
+                    RelativeSizeAxes = Axes.Y,
                     Anchor = Anchor.CentreRight,
                     Origin = Anchor.CentreRight,
                     Child = new Box
@@ -163,9 +163,11 @@ protected override void LoadComplete()
 
         private void updateState()
         {
-            float targetWidth = Width - (ShowDetails.Value ? ButtonsExpandedWidth : ButtonsCollapsedWidth);
+            float buttonAreaWidth = ShowDetails.Value ? ButtonsExpandedWidth : ButtonsCollapsedWidth;
+            float mainAreaWidth = Width - buttonAreaWidth;
 
-            mainArea.ResizeWidthTo(targetWidth, BeatmapCard.TRANSITION_DURATION, Easing.OutQuint);
+            mainArea.ResizeWidthTo(mainAreaWidth, BeatmapCard.TRANSITION_DURATION, Easing.OutQuint);
+            background.ResizeWidthTo(buttonAreaWidth + BeatmapCard.CORNER_RADIUS, BeatmapCard.TRANSITION_DURATION, Easing.OutQuint);
 
             background.FadeColour(downloadTracker.State.Value == DownloadState.LocallyAvailable ? colours.Lime0 : colourProvider.Background3, BeatmapCard.TRANSITION_DURATION, Easing.OutQuint);
             buttons.FadeTo(ShowDetails.Value ? 1 : 0, BeatmapCard.TRANSITION_DURATION, Easing.OutQuint);
```

</details>

which still will look wrong if you squint hard enough but maybe not enough to be unacceptable:

![image](https://user-images.githubusercontent.com/20418176/195670513-44221cde-efdb-44b8-b880-76d19d31de83.png)
![image](https://user-images.githubusercontent.com/20418176/195670590-212ca4e6-ff91-4861-8516-209fea827279.png)

Reply:
> I don't really want to defend this hack too much, but I don't feel that the screenshots in the OP do it justice as to how bad this can look. The worst of it is easily when the card is hovered

Presuming you mean the artifacts on the left side of the background, those are present *regardless* of the existing workaround (at least on my screen, although I can't see how vertical padding would hide it anyways). So this is technically getting out of scope from this PR, but seems like a better and still short workaround that avoids breaking corners so can still apply it here.

Reply:
Bit weird to be touching the hack in this PR and then saying that fixing it further is "out of scope" but ok. I probably wouldn't have bundled it in here in the first place, if I'm honest.

Reply:
> Bit weird to be touching the hack in this PR and then saying that fixing it further is "out of scope" but ok.

Reason I touched it here is because otherwise the hack makes the button look quite awkward with the outer background having a broken corner. Also, I was generally responding to the first paragraph about the screenshots, since the mentioned issue is irrelevant from what the hack achieved, so the screenshots aren't really "not doing it justice".